### ISSUE TITLE
[ENH] Add `CORN` loss function

### ISSUE THREAD

**Issue by JacksonBurns** [ayrna/dlordinal#119]
### Describe the feature or idea you want to propose

Hello, thank you for the great package! I am very new to ordinal classification, so apologies if this is an ill-informed request!

Could the loss function described [in this paper](https://arxiv.org/abs/2111.08851) be added to this package?

### Describe your proposed solution

A reference implementation has been provided [here](https://github.com/Raschka-research-group/coral-pytorch) so it may be possible to just wrap their code in this repo?

**Comment by JacksonBurns** [comment]
@franberchez now it's my turn to apologize! So sorry for taking forever on this, things got busy, but I have made a PR to implement CORN.
